### PR TITLE
Use Request ParamMap instead of req.uri

### PR DESCRIPTION
**Pull request by maheshkelkar**
- since it fetches query param from uri (GET) or contentString (POST)

Also,
- Log an error message if we see multiple ServiceIdentifiers for a
  tuple of (service name and protected flag.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ libraryDependencies ++= Seq(
 
 ```scala
 libraryDependencies ++= Seq(
-  "com.lookout.borderpatrol" %% "[borderpatrol-module]" % "0.1.18-SNAPSHOT"
+  "com.lookout.borderpatrol" %% "[borderpatrol-module]" % "0.1.20-SNAPSHOT"
```
Comment (PeterFCaswell) [inline README.md]: What happened to 0.1.19?

File: auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala [hunk 2]
```diff
@@ -91,14 +91,15 @@ object Keymaster {
   case class KeymasterTransformFilter(oAuth2CodeVerify: OAuth2CodeVerify)(implicit statsReceiver: StatsReceiver)
       extends Filter[BorderRequest, Response, KeymasterIdentifyReq, Response] {
 
-    def transformInternal(req: BorderRequest): Future[InternalAuthCredential] =
+    def transformInternal(req: BorderRequest): Future[InternalAuthCredential] = {
       (for {
-        u <- Helpers.scrubQueryParams(req.req.uri, "username")
-        p <- Helpers.scrubQueryParams(req.req.uri, "password")
+        u <- Helpers.scrubQueryParams(req.req.params, "username")
```
Comment (PeterFCaswell) [inline auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala]: Not something to delay this PR for, but would it be cleaner if you defined a method in BorderRequest, called "params", that simply invoked "req.params"?

File: core/src/main/scala/com/lookout/borderpatrol/util/Helpers.scala [hunk 3]
```diff
@@ -1,35 +1,20 @@
...
-import org.jboss.netty.handler.codec.http.QueryStringDecoder
+import com.twitter.finagle.http.ParamMap
 
-import scala.collection.JavaConverters._
-import scala.util.{Success, Try}
 
 object Helpers {
 
-  /** Parse uri into (path, params) */
+  /** Regular Expression for all special characters from ASCII 0 to 20 */
```
Comment (PeterFCaswell) [inline core/src/main/scala/com/lookout/borderpatrol/util/Helpers.scala]: Is this Octal 0-20 or decimal 0-20? If the latter, what about the characters from NAK to SP?

I really think you need octal 20 or decimal 32.

File: security/src/main/scala/com/lookout/borderpatrol/security/Csrf.scala [hunk 4]
```diff
@@ -40,7 +40,7 @@ object Csrf {
      */
     def verify(req: Request): Boolean =
       (for {
-        str <- req.headerMap.get(header.header) orElse Helpers.scrubQueryParams(req.uri, param.param)
+        str <- req.headerMap.get(header.header) orElse Helpers.scrubQueryParams(req.params, param.param)
```
Comment (PeterFCaswell) [inline security/src/main/scala/com/lookout/borderpatrol/security/Csrf.scala]: "param.param" seems really awkward. Does Param ever have any value other than "_x_border_csrf"?  Wouldn't this be cleaner as "x_crsf.val"?

This won't block the PR, but you should consider it for the future.

**Comment by PeterFCaswell** [comment]
Only things preventing the merge is the special characters you're filtering and the CI build failure.

File: README.md (hunk 1 above)
Comment (maheshkelkar) [inline README.md]: Forgot to update README.md

File: auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala (hunk 2 above)
Comment (maheshkelkar) [inline auth/src/main/scala/com/lookout/borderpatrol/auth/keymaster/Keymaster.scala]: I will add it to the list. Thanks

File: core/src/main/scala/com/lookout/borderpatrol/util/Helpers.scala (hunk 3 above)
Comment (maheshkelkar) [inline core/src/main/scala/com/lookout/borderpatrol/util/Helpers.scala]: I am sorry, thats supposed to be hex. I will correct it

File: security/src/main/scala/com/lookout/borderpatrol/security/Csrf.scala (hunk 4 above)
Comment (maheshkelkar) [inline security/src/main/scala/com/lookout/borderpatrol/security/Csrf.scala]: I will fix that.

**Comment by maheshkelkar** [comment]
@pfc I fixed the comments, and CI is passing now

**Pull request merged**